src/main/java/macrobase/analysis/contextualoutlier/ContextPruning.java

```diff
@@ -0,0 +1,210 @@
+package macrobase.analysis.contextualoutlier;
+
+import java.util.*;
+
+import org.apache.commons.math3.distribution.NormalDistribution;
+import org.apache.commons.math3.stat.inference.TestUtils;
+
+import macrobase.datamodel.Datum;
+
+public class ContextPruning {
```

Comment:
I wanted to have a place to implement all pruning functions for readability. This doesn't have to be a new class. Happy to remove it if you think this is not a good idea.